reconcile/terraform_cloudflare_users.py

```diff
@@ -0,0 +1,327 @@
+    """
+    users: dict[str, dict[str, CloudflareUser]] = {}
+
+    for cf_role in cloudflare_roles or []:
+        if account_name and cf_role.account.name != account_name:
+            continue
+        for access_role in cf_role.access_roles or []:
+            for user in access_role.users:
+                if user.cloudflare_user is not None and (
+                    user.cloudflare_user.split("@")[1]
+                    in (email_domain_allow_list or [])
+                ):
+                    temp = users.get(cf_role.account.name)
+                    if temp is not None:
+                        if temp.get(user.cloudflare_user) is not None:
+                            users[cf_role.account.name][
+                                user.cloudflare_user
+                            ].roles.update(set(cf_role.roles))
+                        else:
+                            users[cf_role.account.name][
+                                user.cloudflare_user
+                            ] = CloudflareUser(
+                                user.cloudflare_user,
+                                cf_role.account.name,
+                                user.org_username,
+                                set(cf_role.roles),
+                            )
+
+                    else:
+                        users[cf_role.account.name] = {
+                            user.cloudflare_user: CloudflareUser(
+                                user.cloudflare_user,
+                                cf_role.account.name,
+                                user.org_username,
+                                set(cf_role.roles),
+                            )
+                        }
+
+    return users
+
+
+def build_external_resource_spec_from_cloudflare_users(
+    cloudflare_users: Mapping[str, Mapping[str, CloudflareUser]]
+) -> Iterable[ExternalResourceSpec]:
+    """
+    This method transforms :param cloudflare_users: into a list of ExternalResourceSpec
+    as TerrascriptCloudflareClient works only with the ExternalResourceSpec.
+    """
+    specs: list[ExternalResourceSpec] = []
+
+    for _, v in cloudflare_users.items():
+        for _, cf_user in v.items():
+            data_source_cloudflare_account_roles = {
+                "identifier": safe_resource_id(cf_user.account_name),
+                "account_id": "${var.account_id}",
+            }
+
+            cloudflare_account_member = {
+                "provider": "cloudflare_account_member",
+                "identifier": safe_resource_id(cf_user.org_username),
+                "email_address": cf_user.email_address,
+                "account_id": "${var.account_id}",
+                "role_ids": [
+                    # I know this is ugly :(
+                    f'%{{ for role in data.cloudflare_account_roles.{safe_resource_id(cf_user.account_name)}.roles ~}}  %{{if role.name == "{each}" ~}}${{role.id}}%{{ endif ~}}  %{{ endfor ~}}'
```

Comment:
Added a comment to explain this better. Hope this helps! :crossed_fingers: 

Regarding testing, I have `test_build_external_resource_spec_from_cloudflare_users` in `test_terraform_cloudflare_users.py` which should exercise the logic of populating `role_ids` with multiple string template for each role. 

This however doesn't exercise if the string template actually works or not, because it is something evaluated by terraform and outside of the unit test boundary. This is where manual testing is currently filling the gap for now. This would be the ideal candidate for integration tests.